### ISSUE TITLE
No Credentials error

### ISSUE THREAD

**Issue by Amandeep-singh11** [pywren/pywren#239]
I am getting below error , while I try to run the executor function

**wrenexec = pywren.default_executor()**

NoCredentialsError                        Traceback (most recent call last)
<ipython-input-3-3a9b42562b70> in <module>()
----> 1 wrenexec = pywren.default_executor()

/usr/local/lib/python3.5/dist-packages/pywren/wren.py in default_executor(**kwargs)
     19 
     20     if executor_str == 'lambda':
---> 21         return lambda_executor(**kwargs)
     22     elif executor_str == 'remote' or executor_str == 'standalone':
     23         return remote_executor(**kwargs)

/usr/local/lib/python3.5/dist-packages/pywren/wren.py in lambda_executor(config, job_max_runtime)
     35     invoker = invokers.LambdaInvoker(AWS_REGION, FUNCTION_NAME)
     36 
---> 37     return Executor(invoker, config, job_max_runtime)
     38 
     39 

/usr/local/lib/python3.5/dist-packages/pywren/executor.py in __init__(self, invoker, config, job_max_runtime)
     38         self.storage_config = wrenconfig.extract_storage_config(self.config)
     39         self.storage = storage.Storage(self.storage_config)
---> 40         self.runtime_meta_info = runtime.get_runtime_info(config['runtime'])
     41 
     42 

/usr/local/lib/python3.5/dist-packages/pywren/runtime.py in get_runtime_info(runtime_config)
      8     Download runtime information from storage at deserialize
      9     """
---> 10     runtime_meta = storage.get_runtime_info(runtime_config)
     11 
     12     if not runtime_valid(runtime_meta):

/usr/local/lib/python3.5/dist-packages/pywren/storage/storage.py in get_runtime_info(runtime_config)
    110     if '.tar.gz' in key:
    111         key = key.replace(".tar.gz", ".meta.json")
--> 112     json_str = handler.get_object(key)
    113     runtime_meta = json.loads(json_str.decode("ascii"))
    114     return runtime_meta

/usr/local/lib/python3.5/dist-packages/pywren/storage/s3_backend.py in get_object(self, key)
     33         """
     34         try:
---> 35             r = self.s3client.get_object(Bucket=self.s3_bucket, Key=key)
     36             data = r['Body'].read()
     37             return data

/usr/local/lib/python3.5/dist-packages/botocore/client.py in _api_call(self, *args, **kwargs)
    312                     "%s() only accepts keyword arguments." % py_operation_name)
    313             # The "self" in this scope is referring to the BaseClient.
--> 314             return self._make_api_call(operation_name, kwargs)
    315 
    316         _api_call.__name__ = str(py_operation_name)

/usr/local/lib/python3.5/dist-packages/botocore/client.py in _make_api_call(self, operation_name, api_params)
    597         else:
    598             http, parsed_response = self._endpoint.make_request(
--> 599                 operation_model, request_dict)
    600 
    601         self.meta.events.emit(

/usr/local/lib/python3.5/dist-packages/botocore/endpoint.py in make_request(self, operation_model, request_dict)
    141         logger.debug("Making request for %s (verify_ssl=%s) with params: %s",
    142                      operation_model, self.verify, request_dict)
--> 143         return self._send_request(request_dict, operation_model)
    144 
    145     def create_request(self, params, operation_model=None):

/usr/local/lib/python3.5/dist-packages/botocore/endpoint.py in _send_request(self, request_dict, operation_model)
    166     def _send_request(self, request_dict, operation_model):
    167         attempts = 1
--> 168         request = self.create_request(request_dict, operation_model)
    169         success_response, exception = self._get_response(
    170             request, operation_model, attempts)

/usr/local/lib/python3.5/dist-packages/botocore/endpoint.py in create_request(self, params, operation_model)
    150                 op_name=operation_model.name)
    151             self._event_emitter.emit(event_name, request=request,
--> 152                                      operation_name=operation_model.name)
    153         prepared_request = self.prepare_request(request)
    154         return prepared_request

/usr/local/lib/python3.5/dist-packages/botocore/hooks.py in emit(self, event_name, **kwargs)
    225                  handlers.
    226         """
--> 227         return self._emit(event_name, kwargs)
    228 
    229     def emit_until_response(self, event_name, **kwargs):

/usr/local/lib/python3.5/dist-packages/botocore/hooks.py in _emit(self, event_name, kwargs, stop_on_response)
    208         for handler in handlers_to_call:
    209             logger.debug('Event %s: calling handler %s', event_name, handler)
--> 210             response = handler(**kwargs)
    211             responses.append((handler, response))
    212             if stop_on_response and response is not None:

/usr/local/lib/python3.5/dist-packages/botocore/signers.py in handler(self, operation_name, request, **kwargs)
     88         # this method is invoked to sign the request.
     89         # Don't call this method directly.
---> 90         return self.sign(operation_name, request)
     91 
     92     def sign(self, operation_name, request, region_name=None,

/usr/local/lib/python3.5/dist-packages/botocore/signers.py in sign(self, operation_name, request, region_name, signing_type, expires_in, signing_name)
    154                     raise e
    155 
--> 156             auth.add_auth(request)
    157 
    158     def _choose_signer(self, operation_name, signing_type, context):

/usr/local/lib/python3.5/dist-packages/botocore/auth.py in add_auth(self, request)
    418         self._region_name = signing_context.get(
    419             'region', self._default_region_name)
--> 420         super(S3SigV4Auth, self).add_auth(request)
    421 
    422     def _modify_request_before_signing(self, request):

/usr/local/lib/python3.5/dist-packages/botocore/auth.py in add_auth(self, request)
    350     def add_auth(self, request):
    351         if self.credentials is None:
--> 352             raise NoCredentialsError
    353         datetime_now = datetime.datetime.utcnow()
    354         request.context['timestamp'] = datetime_now.strftime(SIGV4_TIMESTAMP)

NoCredentialsError: Unable to locate credentials

**Comment by Vaishaal** [comment]
Do you have your aws credentials setup for use from aws CLI?

**Comment by Amandeep-singh11** [comment]
Yeah Vaishaal, I have completed the setup on Ubuntu terminal including setting up the AWS credentials and after that when i am trying to run any program using pywren on jupyter notebook , it gives me this error.

**Comment by Vaishaal** [comment]
did you run pywren-setup?

**Comment by Amandeep-singh11** [comment]
Yes, I run the whole setup on Ubuntu terminal

**Comment by Amandeep-singh11** [comment]
it is returning "Hello World"

**Comment by Amandeep-singh11** [comment]
It is giving me the same error

**NameError: name 'pwex' is not defined**

Does pywren support jupyter notebook ?
I guess this can be the only issue

**Comment by Amandeep-singh11** [comment]
ohh sorry , i got this error

**NoCredentialsError: Unable to locate credentials**

**Comment by ooq** [comment]
This is strange, if `pywren test_function` runs but this does not. It might be related to boto3.
Can you try this?
```
import boto3
s3 = boto3.resource('s3')
```

**Comment by Amandeep-singh11** [comment]
still getting the same error
[Pywren_error.docx](https://github.com/pywren/pywren/files/1891892/Pywren_error.docx)

**Comment by ooq** [comment]
@Amandeep-singh11 Have you tried it from the terminal instead of the notebook?

**Comment by ooq** [comment]
@Amandeep-singh11 This seems to be your python environment failing to find the correct AWS credentials. 
It is somehow contradictory to the fact that `pywren test_function` works. So probably the python environment in which the notebook is started is different from the one you ran `pywren test_function`?
